backend/routes/bus-route.js

```diff
@@ -3,13 +3,37 @@ const router = require('express').Router();
 router.get('/', async (req, res) => {
     let busRoutes = req.busRoutes.get(req.query.ServiceNo);
     if (req.query.Direction) {
-        res.send(busRoutes.get(req.query.Direction));
+        let stopsOfDirection = busRoutes.get(req.query.Direction);
+        let busStops = [];
+        stopsOfDirection.forEach(stop =>{
+            busStops.push(stop.BusStopCode);
+        });
+
+        res.send({
+            ServiceNo : req.query.ServiceNo,
+            Routes : [{
+                OriginCode : stopsOfDirection[0].BusStopCode,
+                DestinationCode : stopsOfDirection[stopsOfDirection.length-1].BusStopCode,
```

Comment:
Oh we are pulling the origin and destination from busServices cache, rather than the first/last busStop for each busRoute